From cfbd8194cf14d4baeea9a160649d39a514bec5ec Mon Sep 17 00:00:00 2001
From: Michael <michael_powell@mentor.com>
Date: Fri, 22 Apr 2016 14:39:04 -0700
Subject: [PATCH] fix assertion of device name to be slightly more lenient

---
 Lib/test/test_ossaudiodev.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/test/test_ossaudiodev.py b/Lib/test/test_ossaudiodev.py
index 7a357a0..89fa33b 100644
--- a/Lib/test/test_ossaudiodev.py
+++ b/Lib/test/test_ossaudiodev.py
@@ -59,7 +59,7 @@ class OSSAudioDevTests(unittest.TestCase):
 
         # Make sure the read-only attributes work.
         self.assertFalse(dsp.closed)
-        self.assertEqual(dsp.name, "/dev/dsp")
+        self.assertRegexpMatches(dsp.name, "/dev/dsp[0-9]?")
         self.assertEqual(dsp.mode, "w", "bad dsp.mode: %r" % dsp.mode)
 
         # And make sure they're really read-only.
-- 
2.4.11

