R600/SI: Add processor type for Mullins.

Signed-off-by: Samuel Li <samuel.li@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
diff -Naur a/lib/Target/R600/Processors.td b/lib/Target/R600/Processors.td
--- a/lib/Target/R600/Processors.td	2013-11-15 05:16:00.000000000 +0530
+++ b/lib/Target/R600/Processors.td	2014-04-26 01:22:39.105011843 +0530
@@ -52,3 +52,4 @@
 def : Proc<"kabini",     SI_Itin, [FeatureSeaIslands]>;
 def : Proc<"kaveri",     SI_Itin, [FeatureSeaIslands]>;
 def : Proc<"hawaii",     SI_Itin, [FeatureSeaIslands]>;
+def : Proc<"mullins",    SI_Itin, [FeatureSeaIslands]>;
