--- git/instrumentation/events/lttng-module/compaction.h.old	2014-12-08 12:40:20.454821330 -0500
+++ git/instrumentation/events/lttng-module/compaction.h	2014-12-08 12:40:55.163040815 -0500
@@ -46,7 +46,7 @@
 	TP_ARGS(nr_scanned, nr_taken)
 )
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,34))
 TRACE_EVENT(mm_compaction_migratepages,
 
 	TP_PROTO(unsigned long nr_all,
@@ -87,7 +87,7 @@
 		__entry->nr_migrated,
 		__entry->nr_failed)
 )
-#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25)) */
+#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,34)) */
 TRACE_EVENT(mm_compaction_migratepages,
 
 	TP_PROTO(unsigned long nr_migrated,
@@ -109,7 +109,7 @@
 		__entry->nr_migrated,
 		__entry->nr_failed)
 )
-#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0)) */
+#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25) || LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,34)) */
 
 #endif /* _TRACE_COMPACTION_H */
 
