diff --git a/init-install-efi.sh b/init-install-efi.sh
index 67ff46d..33622c8
--- init-install-efi.sh
+++ init-install-efi.sh
@@ -16,7 +16,8 @@ swap_ratio=5
 
 # Get a list of hard drives
 hdnamelist=""
-live_dev_name=${1%%/*}
+live_dev_name=`cat /proc/mounts | grep ${1%/} | awk '{print $1}'`
+live_dev_name=${live_dev_name#\/dev/}
 live_dev_name=${live_dev_name%%[0-9]*}
 
 echo "Searching for hard drives ..."
