From 8dc29db401030725ca0201ba03feb6238ec258f7 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 20 Jul 2018 17:58:45 +0500
Subject: [PATCH] configure.ac: obey llvm_prefix if available

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d1c8bb82da..92a83fec4a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2821,7 +2821,11 @@ if test "x$enable_llvm" = xyes; then
 
     if test $LLVM_VERSION_MAJOR -ge 4 -o $LLVM_VERSION_MAJOR -eq 3 -a $LLVM_VERSION_MINOR -ge 9; then
         if test "x$enable_llvm_shared_libs" = xyes; then
-            LLVM_LIBS="`$LLVM_CONFIG --link-shared --libs ${LLVM_COMPONENTS}`"
+            if test "x${llvm_prefix}" == "x"; then
+                LLVM_LIBS="`$LLVM_CONFIG --link-shared --libs ${LLVM_COMPONENTS}`"
+            else
+                LLVM_LIBS="`$LLVM_CONFIG --link-shared --libs ${LLVM_COMPONENTS} --tgtlibdir ${llvm_prefix}`"
+            fi
         else
             dnl Invoking llvm-config with both -libs and --system-libs produces the
             dnl two separate lines - each for the set of libraries.
-- 
2.11.1

