diff --git a/src/gallium/targets/r600/omx/Makefile.am b/src/gallium/targets/r600/omx/Makefile.am
index 7660a8d..fee1572 100644
--- a/src/gallium/targets/r600/omx/Makefile.am
+++ b/src/gallium/targets/r600/omx/Makefile.am
@@ -41,6 +41,7 @@ libomx_r600_la_SOURCES = \
 
 libomx_r600_la_LDFLAGS = \
 	-module \
+	-avoid-version \
 	-export-symbols-regex $(EXPORTS) \
 	-shared \
 	-no-undefined
diff --git a/src/gallium/targets/radeonsi/omx/Makefile.am b/src/gallium/targets/radeonsi/omx/Makefile.am
index 5eea303..0af64bc 100644
--- a/src/gallium/targets/radeonsi/omx/Makefile.am
+++ b/src/gallium/targets/radeonsi/omx/Makefile.am
@@ -41,6 +41,7 @@ libomx_radeonsi_la_SOURCES = \
 
 libomx_radeonsi_la_LDFLAGS = \
 	-module \
+	-avoid-version \
 	-export-symbols-regex $(EXPORTS) \
 	-shared \
 	-no-undefined
