From 5cf5f62f912c3cd178561ad164f1f51f980e96a9 Mon Sep 17 00:00:00 2001
From: Samuel Li <samuel.li@amd.com>
Date: Thu, 17 Apr 2014 15:17:28 -0400
Subject: [PATCH 1/2] radeon: add support for Mullins.

Signed-off-by: Samuel Li <samuel.li@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 man/radeon.man     | 3 +++
 src/radeon_probe.h | 1 +
 2 files changed, 4 insertions(+)

diff --git a/man/radeon.man b/man/radeon.man
index f6a825d..65b880c 100644
--- a/man/radeon.man
+++ b/man/radeon.man
@@ -208,6 +208,9 @@ KABINI APUs
 .TP 12
 .B HAWAII
 Radeon R9 series
+.TP 12
+.B MULLINS
+MULLINS APUs
 .PD
 .SH CONFIGURATION DETAILS
 Please refer to __xconfigfile__(__filemansuffix__) for general configuration
diff --git a/src/radeon_probe.h b/src/radeon_probe.h
index 9500189..3fe4644 100644
--- a/src/radeon_probe.h
+++ b/src/radeon_probe.h
@@ -116,6 +116,7 @@ typedef enum {
     CHIP_FAMILY_KAVERI,
     CHIP_FAMILY_KABINI,
     CHIP_FAMILY_HAWAII,
+    CHIP_FAMILY_MULLINS,
     CHIP_FAMILY_LAST
 } RADEONChipFamily;
 
-- 
1.8.3.1

