From 2ad5ea780b3cca83ae4f531ae0b4159e802ef825 Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan@agner.ch>
Date: Sat, 19 Dec 2015 21:52:59 -0800
Subject: [PATCH 014/117] tests: add fsl-dcu-drm to modules

Signed-off-by: Stefan Agner <stefan@agner.ch>
Reviewed-by: Emil Velikov <emil.l.velikov@gmail.com>
---
 tests/util/kms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/util/kms.c b/tests/util/kms.c
index 57b0191..dcd5a8e 100644
--- a/tests/util/kms.c
+++ b/tests/util/kms.c
@@ -139,6 +139,7 @@ static const char * const modules[] = {
 	"imx-drm",
 	"rockchip",
 	"atmel-hlcdc",
+	"fsl-dcu-drm",
 };
 
 int util_open(const char *device, const char *module)
-- 
2.7.4

