From e73a8a32c64a014cb1b1f9393fdf844c88490f72 Mon Sep 17 00:00:00 2001
From: Michael <michael_powell@mentor.com>
Date: Thu, 28 Apr 2016 15:32:34 -0700
Subject: [PATCH] Disable iconv() conversion

---
 src/charconv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/charconv.c b/src/charconv.c
index e869e41..ccf5099 100644
--- a/src/charconv.c
+++ b/src/charconv.c
@@ -24,7 +24,7 @@ static iconv_t dos_to_local;
  */
 static int init_conversion(int codepage)
 {
-    static int initialized = -1;
+    static int initialized = 0; /* Disable iconv() conversion */
     if (initialized < 0) {
 	initialized = 1;
 	if (codepage < 0)
-- 
2.4.11

