--- CodeXL/Components/GpuProfiling/Build/CommonTargets.mk.orig	2017-01-19 15:56:10.300696899 -0500
+++ CodeXL/Components/GpuProfiling/Build/CommonTargets.mk	2017-01-19 16:05:12.473012993 -0500
@@ -39,7 +39,7 @@
 
 # source file targets
 
-BUILD_SRC=$(CC) $(CFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@
+BUILD_SRC=mkdir -p $(@D); $(CC) $(CFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@
 
 $(OBJ_DIR)/%.o: %.cpp
 	$(BUILD_SRC)
