From e7a9925b4d85904cf4511d172bcdf05d27d576f3 Mon Sep 17 00:00:00 2001
From: Sanjay R Mehta <sanju.mehta@amd.com>
Date: Mon, 4 Apr 2016 19:35:54 +0530
Subject: [PATCH 2/2] yocto amd i2c dev add calls to enable and disable IMC
 from fetching BIOS code

---
 drivers/i2c/i2c-dev.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/i2c/i2c-dev.c b/drivers/i2c/i2c-dev.c
index 71c7a39..3072f88 100644
--- a/drivers/i2c/i2c-dev.c
+++ b/drivers/i2c/i2c-dev.c
@@ -34,6 +34,7 @@
 #include <linux/i2c-dev.h>
 #include <linux/jiffies.h>
 #include <linux/uaccess.h>
+#include <linux/amd_imc.h>
 
 /*
  * An i2c_dev represents an i2c_adapter ... an I2C or SMBus master, not a
@@ -513,6 +514,8 @@ static int i2cdev_open(struct inode *inode, struct file *file)
 	client->adapter = adap;
 	file->private_data = client;
 
+	amd_imc_enter_scratch_ram();
+
 	return 0;
 }
 
@@ -524,6 +527,8 @@ static int i2cdev_release(struct inode *inode, struct file *file)
 	kfree(client);
 	file->private_data = NULL;
 
+	amd_imc_exit_scratch_ram();
+
 	return 0;
 }
 
-- 
1.9.1

