From e0f470758a67e9ac1b7007238211d4124e6f6055 Mon Sep 17 00:00:00 2001
From: Jammy Zhou <Jammy.Zhou@amd.com>
Date: Mon, 18 May 2015 19:14:56 +0800
Subject: [PATCH 07/67] amdgpu: fix the number of IB size enums
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jammy Zhou <Jammy.Zhou@amd.com>
Reviewed-by: Michel Dänzer <michel.daenzer@amd.com>
Reviewed-by: Christian König <christian.koenig@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 amdgpu/amdgpu.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/amdgpu/amdgpu.h b/amdgpu/amdgpu.h
index c48af92..e99bada 100644
--- a/amdgpu/amdgpu.h
+++ b/amdgpu/amdgpu.h
@@ -104,15 +104,15 @@ enum amdgpu_bo_handle_type {
  * \sa amdgpu_cs_alloc_ib()
  */
 enum amdgpu_cs_ib_size {
-	amdgpu_cs_ib_size_4K	= 1,
-	amdgpu_cs_ib_size_16K	= 2,
-	amdgpu_cs_ib_size_32K	= 3,
-	amdgpu_cs_ib_size_64K	= 4,
-	amdgpu_cs_ib_size_128K	= 5
+	amdgpu_cs_ib_size_4K	= 0,
+	amdgpu_cs_ib_size_16K	= 1,
+	amdgpu_cs_ib_size_32K	= 2,
+	amdgpu_cs_ib_size_64K	= 3,
+	amdgpu_cs_ib_size_128K	= 4
 };
 
 /** The number of different IB sizes */
-#define AMDGPU_CS_IB_SIZE_NUM 6
+#define AMDGPU_CS_IB_SIZE_NUM 5
 
 
 /*--------------------------------------------------------------------------*/
-- 
1.9.1

