From 2004d38559970dd9d59a6cb0d834fdaddbc65dd3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Ol=C5=A1=C3=A1k?= <marek.olsak@amd.com>
Date: Thu, 4 Jun 2015 12:47:22 +0200
Subject: [PATCH 32/67] amdgpu: allow exporting KMS handles with render nodes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Christian König <christian.koenig@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 amdgpu/amdgpu_bo.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/amdgpu/amdgpu_bo.c b/amdgpu/amdgpu_bo.c
index f797234..5ac78de 100644
--- a/amdgpu/amdgpu_bo.c
+++ b/amdgpu/amdgpu_bo.c
@@ -303,9 +303,11 @@ int amdgpu_bo_export(amdgpu_bo_handle bo,
 		return 0;
 
 	case amdgpu_bo_handle_type_kms:
-		r = amdgpu_bo_export_flink(bo);
-		if (r)
-			return r;
+		/* don't check the return value, because exporting KMS
+		 * handles should work with render nodes too, which don't
+		 * support flink exports
+		 */
+		amdgpu_bo_export_flink(bo);
 
 		amdgpu_add_handle_to_table(bo);
 		*shared_handle = bo->handle;
-- 
1.9.1

