From a06645866342f1cbd4d0be9258fd48d2a8006c97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20K=C3=B6nig?= <christian.koenig@amd.com>
Date: Tue, 30 Jun 2015 15:46:29 +0200
Subject: [PATCH 47/67] gitignore: add some generated amdgpu files
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Christian König <christian.koenig@amd.com>
Reviewed-by: Jammy Zhou <Jammy.Zhou@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index b876973..79b2ac1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -55,6 +55,7 @@ libdrm_radeon.pc
 libdrm_omap.pc
 libdrm_exynos.pc
 libdrm_freedreno.pc
+libdrm_amdgpu.pc
 libkms.pc
 libtool
 ltmain.sh
@@ -73,6 +74,7 @@ stamp-h1
 tdfx.kld
 via.kld
 tests/auth
+tests/amdgpu/amdgpu_test
 tests/dristat
 tests/drmsl
 tests/drmstat
-- 
1.9.1

