From 8ec69a8dca952a7c62b7c5dc0137813d5566786d Mon Sep 17 00:00:00 2001
From: Jammy Zhou <Jammy.Zhou@amd.com>
Date: Mon, 13 Jul 2015 22:43:29 +0800
Subject: [PATCH 59/67] tests/amdgpu: remove the duplicate IB allocation for
 VCE test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jammy Zhou <Jammy.Zhou@amd.com>
Reviewed-by: Christian König <christian.koenig@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 tests/amdgpu/vce_tests.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/tests/amdgpu/vce_tests.c b/tests/amdgpu/vce_tests.c
index 1d2cd40..1c04044 100644
--- a/tests/amdgpu/vce_tests.c
+++ b/tests/amdgpu/vce_tests.c
@@ -156,14 +156,6 @@ static int submit(unsigned ndw, unsigned ip)
 	if (r)
 		return r;
 
-	r = amdgpu_bo_alloc_and_map(device_handle, IB_SIZE, 4096,
-				    AMDGPU_GEM_DOMAIN_GTT, 0,
-				    &ib_handle, (void**)&ib_cpu,
-				    &ib_mc_address);
-	if (r)
-		return r;
-	resources[num_resources-1] = ib_handle;
-
 	fence_status.context = context_handle;
 	fence_status.ip_type = ip;
 
-- 
1.9.1

