From b573ccbe2b3fcf15ccea77782a2cd3b62c1a7405 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20K=C3=B6nig?= <christian.koenig@amd.com>
Date: Wed, 15 Jul 2015 10:56:20 +0200
Subject: [PATCH 60/67] amdgpu: fix bs buffer size for vce test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Christian König <christian.koenig@amd.com>
Reviewed-by: Jammy Zhou <Jammy.Zhou@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 tests/amdgpu/vce_tests.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/amdgpu/vce_tests.c b/tests/amdgpu/vce_tests.c
index 1c04044..1039890 100644
--- a/tests/amdgpu/vce_tests.c
+++ b/tests/amdgpu/vce_tests.c
@@ -363,7 +363,7 @@ static void check_result(struct amdgpu_vce_encode *enc)
 
 static void amdgpu_cs_vce_encode(void)
 {
-	uint32_t vbuf_size, bs_size = 0x2000, cpb_size;
+	uint32_t vbuf_size, bs_size = 0x154000, cpb_size;
 	int i, r;
 
 	vbuf_size = enc.width * enc.height * 1.5;
-- 
1.9.1

