diff --git a/configure.ac b/configure.ac
index 12dedd6..0776938 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1012,6 +1012,19 @@ llvm_add_target() {
     fi
 }
 
+# Strip all the native bits from ldflags
+strip_unwanted_llvm_ldflags() {
+    flags=""
+    for flag in `$1`; do
+        case "$flag" in
+            *native*) ;;
+            *) flags="$flags $flag"
+                ;;
+        esac
+    done
+    echo $flags
+}
+
 # Call this inside ` ` to get the return value.
 # $1 is the llvm-config command with arguments.
 strip_unwanted_llvm_flags() {
@@ -2661,7 +2674,7 @@ dnl
 if test "x$enable_llvm" = xyes; then
     DEFINES="${DEFINES} -DHAVE_LLVM=0x0$LLVM_VERSION_INT -DMESA_LLVM_VERSION_PATCH=$LLVM_VERSION_PATCH"
 
-    LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
+    LLVM_LDFLAGS=`strip_unwanted_llvm_ldflags "$LLVM_CONFIG --ldflags"`
     LLVM_CFLAGS=$LLVM_CPPFLAGS   # CPPFLAGS seem to be sufficient
     LLVM_CXXFLAGS=`strip_unwanted_llvm_flags "$LLVM_CONFIG --cxxflags"`
 
