From 7fd5e7ac8d476bc7538eb6e11875cb87233f0bdc Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 14 Mar 2018 11:27:26 +0500
Subject: [PATCH 1/2] crypto/ccp: fix shared libs build

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 drivers/crypto/ccp/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/crypto/ccp/Makefile b/drivers/crypto/ccp/Makefile
index 1475a6c..4ba8757 100644
--- a/drivers/crypto/ccp/Makefile
+++ b/drivers/crypto/ccp/Makefile
@@ -19,6 +19,7 @@ LDLIBS += -lcrypto
 LDLIBS += -lrte_eal -lrte_mbuf -lrte_mempool -lrte_ring
 LDLIBS += -lrte_cryptodev
 LDLIBS += -lrte_pci -lrte_bus_pci
+LDLIBS += -lrte_bus_vdev
 
 # versioning export map
 EXPORT_MAP := rte_pmd_ccp_version.map
-- 
2.11.1

