From b4fd269981cf81eb82e9d56e83c9bd1bc1d00609 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 14 Mar 2018 11:28:18 +0500
Subject: [PATCH 2/2] net/axgbe: fix shared libs build

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 drivers/net/axgbe/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/axgbe/Makefile b/drivers/net/axgbe/Makefile
index e1e0306..de6bd69 100644
--- a/drivers/net/axgbe/Makefile
+++ b/drivers/net/axgbe/Makefile
@@ -11,6 +11,10 @@ LIB = librte_pmd_axgbe.a
 CFLAGS += -O3
 CFLAGS += $(WERROR_FLAGS)
 
+LDLIBS += -lrte_eal -lrte_mempool
+LDLIBS += -lrte_pci -lrte_bus_pci
+LDLIBS += -lrte_ethdev
+
 EXPORT_MAP := rte_pmd_axgbe_version.map
 
 LIBABIVER := 1
-- 
2.11.1

