From 233ba185542405bc38ecb01d1dd90881056291cf Mon Sep 17 00:00:00 2001
From: Ahsan Hussain <ahsan_hussain@mentor.com>
Date: Tue, 19 Sep 2017 13:59:27 +0500
Subject: [PATCH 1/1] CMakeLists: add include paths for xcb and Xlib

---
 vktrace/src/vktrace_viewer/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/vktrace/src/vktrace_viewer/CMakeLists.txt b/vktrace/src/vktrace_viewer/CMakeLists.txt
index dfe2a48..7fbfa12 100644
--- a/vktrace/src/vktrace_viewer/CMakeLists.txt
+++ b/vktrace/src/vktrace_viewer/CMakeLists.txt
@@ -22,6 +22,9 @@ else()
 find_package(Threads REQUIRED)
 find_package(X11 REQUIRED)
 
+find_path(XCB_INCLUDE_DIR xcb/xcb.h DOC "Path to xcb/xcb.h")
+find_path(X11_XLIB_INCLUDE_DIR X11/Xlib.h DOC "Path to X11/Xlib.h")
+
 require_pthreads()
 
 include_directories(
@@ -33,6 +36,8 @@ include_directories(
     ${SRC_DIR}/vktrace_viewer
     ${CMAKE_CURRENT_BINARY_DIR}
     ${Qt5Widgets_INCLUDE_DIRS}
+    ${X11_XLIB_INCLUDE_DIR}
+    ${XCB_INCLUDE_DIR}
 )
 
 set(SRC_LIST
-- 
2.11.1

