From 37ab960eef4b96785906487cbb11bdf08a4e42b8 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 26 Dec 2017 09:47:22 -0500
Subject: [PATCH 1/4] Update kvm instrumentation for 4.15

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index a8b3e9a..c01772c 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -84,6 +84,22 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 	{ KVM_TRACE_MMIO_READ, "read" }, \
 	{ KVM_TRACE_MMIO_WRITE, "write" }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,34))
+
+LTTNG_TRACEPOINT_EVENT(kvm_mmio,
+	TP_PROTO(int type, int len, u64 gpa, void *val),
+	TP_ARGS(type, len, gpa, val),
+
+	TP_FIELDS(
+		ctf_integer(u32, type, type)
+		ctf_integer(u32, len, len)
+		ctf_integer(u64, gpa, gpa)
+		ctf_sequence_hex(unsigned char, val, val, u32, len)
+	)
+)
+
+#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	TP_PROTO(int type, int len, u64 gpa, u64 val),
 	TP_ARGS(type, len, gpa, val),
@@ -96,6 +112,8 @@ LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	)
 )
 
+#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,34))
 
 #define kvm_fpu_load_symbol	\
-- 
2.11.1

