diff -rub mesa-demos-8.2.0.orig/src/Makefile.am mesa-demos-8.2.0/src/Makefile.am
--- mesa-demos-8.2.0.orig/src/Makefile.am	2014-11-26 19:02:35.170565888 -0500
+++ mesa-demos-8.2.0/src/Makefile.am	2014-11-26 19:03:26.102946753 -0500
@@ -22,7 +22,12 @@
 # Authors:
 #    Eric Anholt <eric@anholt.net>
 
+if HAVE_GLEW
+UTIL = util
+endif
+
 SUBDIRS = \
+	$(UTIL) \
 	data \
 	demos \
 	egl \
@@ -46,6 +51,5 @@
 SUBDIRS += \
 	vp \
 	vpglsl \
-	util \
 	trivial
 endif
