diff -rub mesa-demos-8.2.0-orig/configure.ac mesa-demos-8.2.0/configure.ac
--- mesa-demos-8.2.0-orig/configure.ac	2014-12-10 21:29:43.975686890 -0500
+++ mesa-demos-8.2.0/configure.ac	2014-12-10 21:31:27.284438993 -0500
@@ -118,7 +118,7 @@
 )
 
 if test "x$enable_glu" = xyes; then
-    PKG_CHECK_MODULES(GLU, [glu], [],
+    PKG_CHECK_MODULES(GLU, [glu],
                      [AC_CHECK_HEADER([GL/glu.h],
                                       [],
                                       AC_MSG_ERROR([GLU not found]))
@@ -126,7 +126,8 @@
                                    [gluBeginCurve],
                                    [GLU_LIBS=-lGLU
 				    glu_enabled=yes],
-                                   AC_MSG_ERROR([GLU required])) ])
+                                   AC_MSG_ERROR([GLU required])) ],
+                      [])
 
     DEMO_CFLAGS="$DEMO_CFLAGS $GLU_CFLAGS"
     DEMO_LIBS="$DEMO_LIBS $GLU_LIBS"
diff -rub mesa-demos-8.2.0-orig/src/Makefile.am mesa-demos-8.2.0/src/Makefile.am
--- mesa-demos-8.2.0-orig/src/Makefile.am	2014-12-10 21:29:43.935686618 -0500
+++ mesa-demos-8.2.0/src/Makefile.am	2014-12-10 21:30:18.395937483 -0500
@@ -22,7 +22,12 @@
 # Authors:
 #    Eric Anholt <eric@anholt.net>
 
+if HAVE_GLEW
+UTIL = util
+endif
+
 SUBDIRS = \
+	$(UTIL) \
 	data \
 	demos \
 	egl \
@@ -46,6 +51,5 @@
 SUBDIRS += \
 	vp \
 	vpglsl \
-	util \
 	trivial
 endif
