diff --git a/src/Makefile.am b/src/Makefile.am
index 754c47c..b930e06 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -26,18 +26,7 @@ SUBDIRS = \
 	data \
 	demos \
 	egl \
-	fp \
-	fpglsl \
-	glsl \
-	gs \
-	objviewer \
-	osdemos \
-	perf \
 	rbug \
-	redbook \
-	samples \
-	slang \
-	tests \
 	tools \
 	wgl \
 	xdemos
@@ -47,5 +36,16 @@ SUBDIRS += \
 	vp \
 	vpglsl \
 	util \
-	trivial
+	trivial \
+	fp \
+	fpglsl \
+	glsl \
+	gs \
+	objviewer \
+	osdemos \
+	perf \
+	redbook \
+	slang \
+	samples \
+	tests
 endif
