From 2ee96475fe61bbc8a22175e774cc3b9d3627c4a9 Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew_moseley@mentor.com>
Date: Fri, 25 Jul 2014 18:28:09 -0400
Subject: [PATCH] xhci: Enable XHCI_TRUST_TX_LENGTH quirk for AMD devices.

Signed-off-by: Drew Moseley <drew_moseley@mentor.com>
---
 drivers/usb/host/xhci-pci.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index f34b42e..ff2d876 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -38,6 +38,9 @@
 #define PCI_DEVICE_ID_INTEL_LYNXPOINT_XHCI	0x8c31
 #define PCI_DEVICE_ID_INTEL_LYNXPOINT_LP_XHCI	0x9c31
 
+#define PCI_VENDOR_ID_AMD	0x1022
+#define PCI_DEVICE_ID_AMD_USB_HCI	0x7814
+
 static const char hcd_name[] = "xhci_hcd";
 
 /* called after powerup, by probe or system-pm "wakeup" */
@@ -90,6 +93,13 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
 				"has broken MSI implementation",
 				pdev->revision);
 		xhci->quirks |= XHCI_TRUST_TX_LENGTH;
+	} else if (pdev->vendor == PCI_VENDOR_ID_AMD &&
+			   pdev->device == PCI_DEVICE_ID_AMD_USB_HCI) {
+		xhci->quirks |= XHCI_TRUST_TX_LENGTH;
+		xhci_dbg_trace(xhci, trace_xhci_dbg_quirks,
+				"QUIRK: AMD revision %u "
+				"needs quirk XHCI_TRUST_TX_LENGTH",
+				pdev->revision);
 	}
 
 	if (pdev->vendor == PCI_VENDOR_ID_NEC)
-- 
1.9.1

