From 39bff6c13832995b0cd48453cf705a13d906f9d5 Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew_moseley@mentor.com>
Date: Fri, 11 Jul 2014 19:00:56 -0400
Subject: [PATCH] Adjust gstomx.conf for Radeon

Signed-off-by: Drew Moseley <drew_moseley@mentor.com>
---
 config/bellagio/gstomx.conf | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/config/bellagio/gstomx.conf b/config/bellagio/gstomx.conf
index 78f8e7d..bc26507 100644
--- a/config/bellagio/gstomx.conf
+++ b/config/bellagio/gstomx.conf
@@ -10,11 +10,10 @@ hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-chang
 [omxh264dec]
 type-name=GstOMXH264Dec
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.video_decoder.avc
+component-name=OMX.radeonsi.video_decoder.avc
 rank=256
 in-port-index=0
 out-port-index=1
-hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-changed-port-0-to-1
 
 [omxmpeg4videoenc]
 type-name=GstOMXMPEG4VideoEnc
@@ -33,3 +32,19 @@ rank=0
 in-port-index=0
 out-port-index=1
 hacks=event-port-settings-changed-ndata-parameter-swap
+
+[omxmpeg2dec]
+type-name=GstOMXMPEG2VideoDec
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.radeonsi.video_decoder.mpeg2
+rank=256
+in-port-index=0
+out-port-index=1
+
+[omxh264enc]
+type-name=GstOMXH264Enc
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.radeonsi.video_encoder.avc
+rank=256
+in-port-index=0
+out-port-index=1
-- 
1.9.1

