From 1091020cfd26089583d3f83f21c26bba88d5e8d0 Mon Sep 17 00:00:00 2001
From: Arindam Nath <arindam.nath@amd.com>
Date: Fri, 17 Apr 2015 18:26:10 +0530
Subject: [PATCH 1/1] adjust gstomx.conf for mesa-omx

Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 config/bellagio/gstomx.conf | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/config/bellagio/gstomx.conf b/config/bellagio/gstomx.conf
index 78f8e7d..bc26507 100644
--- a/config/bellagio/gstomx.conf
+++ b/config/bellagio/gstomx.conf
@@ -10,11 +10,10 @@ hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-chang
 [omxh264dec]
 type-name=GstOMXH264Dec
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.video_decoder.avc
+component-name=OMX.mesa.video_decoder.avc
 rank=256
 in-port-index=0
 out-port-index=1
-hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-changed-port-0-to-1
 
 [omxmpeg4videoenc]
 type-name=GstOMXMPEG4VideoEnc
@@ -33,3 +32,19 @@ rank=0
 in-port-index=0
 out-port-index=1
 hacks=event-port-settings-changed-ndata-parameter-swap
+
+[omxmpeg2dec]
+type-name=GstOMXMPEG2VideoDec
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.mesa.video_decoder.mpeg2
+rank=256
+in-port-index=0
+out-port-index=1
+
+[omxh264enc]
+type-name=GstOMXH264Enc
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.mesa.video_encoder.avc
+rank=256
+in-port-index=0
+out-port-index=1
-- 
1.9.1

