From b9e5a41ec439d9ccbe18aaa6b7437ca2be82498a Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 6 Apr 2017 15:45:12 +0500
Subject: [PATCH] adjust gstomx.conf for mesa-omx

Signed-off-by: Arindam Nath <arindam.nath@amd.com>
Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 config/bellagio/gstomx.conf | 25 ++++++++++++++++++++-----
 1 file changed, 20 insertions(+), 5 deletions(-)

diff --git a/config/bellagio/gstomx.conf b/config/bellagio/gstomx.conf
index 5ca8ba6..48c243d 100644
--- a/config/bellagio/gstomx.conf
+++ b/config/bellagio/gstomx.conf
@@ -1,7 +1,7 @@
 [omxmpeg4videodec]
 type-name=GstOMXMPEG4VideoDec
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.video_decoder.mpeg4
+component-name=OMX.mesa.video_decoder.mpeg4
 rank=257
 in-port-index=0
 out-port-index=1
@@ -10,16 +10,15 @@ hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-chang
 [omxh264dec]
 type-name=GstOMXH264Dec
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.video_decoder.avc
+component-name=OMX.mesa.video_decoder.avc
 rank=257
 in-port-index=0
 out-port-index=1
-hacks=event-port-settings-changed-ndata-parameter-swap;event-port-settings-changed-port-0-to-1
 
 [omxmpeg4videoenc]
 type-name=GstOMXMPEG4VideoEnc
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.video_encoder.mpeg4
+component-name=OMX.mesa.video_encoder.mpeg4
 rank=0
 in-port-index=0
 out-port-index=1
@@ -28,8 +27,24 @@ hacks=event-port-settings-changed-ndata-parameter-swap;video-framerate-integer;s
 [omxaacenc]
 type-name=GstOMXAACEnc
 core-name=/usr/local/lib/libomxil-bellagio.so.0
-component-name=OMX.st.audio_encoder.aac
+component-name=OMX.mesa.audio_encoder.aac
 rank=0
 in-port-index=0
 out-port-index=1
 hacks=event-port-settings-changed-ndata-parameter-swap
+
+[omxmpeg2dec]
+type-name=GstOMXMPEG2VideoDec
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.mesa.video_decoder.mpeg2
+rank=256
+in-port-index=0
+out-port-index=1
+
+[omxh264enc]
+type-name=GstOMXH264Enc
+core-name=/usr/local/lib/libomxil-bellagio.so.0
+component-name=OMX.mesa.video_encoder.avc
+rank=256
+in-port-index=0
+out-port-index=1
-- 
2.11.1

