From c321d1666fd107d5e3a673789ddf5b9db2b831ec Mon Sep 17 00:00:00 2001
From: Leo Liu <leo.liu@amd.com>
Date: Fri, 22 Aug 2014 12:37:55 -0400
Subject: [PATCH 14/14] gstomxvideoenc: reduce shutdown timeout for tunnelling

Signed-off-by: Leo Liu <leo.liu@amd.com>
---
 omx/gstomxvideoenc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxvideoenc.c b/omx/gstomxvideoenc.c
index 83c68fe..f42ec13 100644
--- a/omx/gstomxvideoenc.c
+++ b/omx/gstomxvideoenc.c
@@ -528,7 +528,10 @@ gst_omx_video_enc_shutdown (GstOMXVideoEnc * self)
       gst_omx_port_set_enabled (self->enc_in_port, FALSE);
     gst_omx_port_deallocate_buffers (self->enc_out_port);
     if (state > OMX_StateLoaded)
-      gst_omx_component_get_state (self->enc, 5 * GST_SECOND);
+      if (!self->enc_in_port->tunneled)
+        gst_omx_component_get_state (self->enc, 5 * GST_SECOND);
+      else
+        gst_omx_component_get_state (self->enc, 1 * GST_SECOND);
   }
 
   return TRUE;
-- 
1.9.1

