From 0b42c7cafb1f0f836c26dd5f0309d400a3b33740 Mon Sep 17 00:00:00 2001
From: Leo Liu <leo.liu@amd.com>
Date: Mon, 28 Apr 2014 15:44:55 -0400
Subject: [PATCH] drm/radeon: add Mulins VCE support

Signed-off-by: Leo Liu <leo.liu@amd.com>
diff -Naur a/drivers/gpu/drm/radeon/radeon_vce.c b/drivers/gpu/drm/radeon/radeon_vce.c
--- a/drivers/gpu/drm/radeon/radeon_vce.c	2014-04-29 15:47:47.888192142 +0530
+++ b/drivers/gpu/drm/radeon/radeon_vce.c	2014-04-29 15:50:44.172184659 +0530
@@ -67,6 +67,7 @@
 	case CHIP_BONAIRE:
 	case CHIP_KAVERI:
 	case CHIP_KABINI:
+	case CHIP_MULLINS:
 		fw_name = FIRMWARE_BONAIRE;
 		break;
 
