From 87edcb996897825ef5dbbf6a614e1f07cc5edb5c Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 4 Sep 2018 17:10:08 +0500
Subject: [PATCH 2/2] layersvt: drop device_sim as we don't have jsoncpp for
 now

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 layersvt/CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/layersvt/CMakeLists.txt b/layersvt/CMakeLists.txt
index 9b780ce4d..42fd3c471 100644
--- a/layersvt/CMakeLists.txt
+++ b/layersvt/CMakeLists.txt
@@ -46,7 +46,6 @@ set(LAYER_JSON_FILES
     VkLayer_api_dump
     VkLayer_monitor
     VkLayer_screenshot
-    VkLayer_device_simulation
     )
 
 set(VK_LAYER_RPATH /usr/lib/x86_64-linux-gnu/vulkan/layer:/usr/lib/i386-linux-gnu/vulkan/layer)
@@ -135,7 +134,6 @@ include_directories(
     ${CMAKE_BINARY_DIR}
     ${CMAKE_BINARY_DIR}/${V_LVL_RELATIVE_LOCATION}
     ${PROJECT_SOURCE_DIR}/../glslang/SPIRV
-    ${JSONCPP_INCLUDE_DIR}
     ${X11_XLIB_INCLUDE_DIR}
 )
 
@@ -168,6 +166,5 @@ run_vk_xml_generate(api_dump_generator.py api_dump_html.h)
 
 add_vk_layer(monitor monitor.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_table.cpp)
 add_vk_layer(screenshot screenshot.cpp screenshot_parsing.h screenshot_parsing.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_table.cpp)
-add_vk_layer(device_simulation device_simulation.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_table.cpp ${JSONCPP_SOURCE_DIR}/jsoncpp.cpp)
 add_vk_layer(api_dump api_dump.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_table.cpp)
 
-- 
2.11.1

