From 526f0b037740c76495dab36e9f2e4d7672dcbac7 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Mon, 10 Apr 2017 15:32:43 -0400
Subject: [PATCH 1195/2094] drm/amdgpu: fix spelling in header comment
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Christian König <christian.koenig@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 include/uapi/drm/amdgpu_drm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/drm/amdgpu_drm.h b/include/uapi/drm/amdgpu_drm.h
index 3243409..d76d525 100644
--- a/include/uapi/drm/amdgpu_drm.h
+++ b/include/uapi/drm/amdgpu_drm.h
@@ -299,7 +299,7 @@ union drm_amdgpu_gem_wait_idle {
 struct drm_amdgpu_wait_cs_in {
 	/* Command submission handle
          * handle equals 0 means none to wait for
-         * handle equal ~0ull meanas wait for the latest sequence number
+         * handle equals ~0ull means wait for the latest sequence number
          */
 	__u64 handle;
 	/** Absolute timeout to wait */
-- 
2.7.4

