From c67b2175cde0859f48a2134dcea8921f65df53b0 Mon Sep 17 00:00:00 2001
From: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
Date: Fri, 8 Dec 2017 10:00:33 +0530
Subject: [PATCH] Enabled DMA flags in eMMC driver.

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>

Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/mmc/host/sdhci-acpi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/sdhci-acpi.c b/drivers/mmc/host/sdhci-acpi.c
index 0944a8c..232bd8e 100755
--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -398,8 +398,8 @@ static const struct sdhci_acpi_slot sdhci_acpi_slot_qcom_sd = {
 static const struct sdhci_acpi_slot sdhci_acpi_slot_amd_emmc = {
 	.chip   = &sdhci_acpi_chip_amd,
 	.caps   = MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE |  MMC_CAP_HW_RESET,
-	//.quirks = SDHCI_QUIRK_32BIT_DMA_ADDR | SDHCI_QUIRK_32BIT_DMA_SIZE | SDHCI_QUIRK_32BIT_ADMA_SIZE,
-    .quirks  =  SDHCI_QUIRK_BROKEN_DMA | SDHCI_QUIRK_BROKEN_ADMA,
+	.quirks = SDHCI_QUIRK_32BIT_DMA_ADDR | SDHCI_QUIRK_32BIT_DMA_SIZE |
+				SDHCI_QUIRK_32BIT_ADMA_SIZE,
 	.probe_slot     = sdhci_acpi_emmc_amd_probe_slot,
 };
 struct sdhci_acpi_uid_slot {
-- 
2.7.4

