From 4c32b1abaf731b6c0d48c2879fad7be4870c2bac Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Fri, 2 Mar 2018 15:24:54 +0530
Subject: [PATCH 3298/5855] IRQ polarity configuration moved to BIOS

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 drivers/acpi/resource.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/acpi/resource.c b/drivers/acpi/resource.c
index 0990e20..56241eb 100755
--- a/drivers/acpi/resource.c
+++ b/drivers/acpi/resource.c
@@ -412,11 +412,6 @@ static void acpi_dev_get_irqresource(struct resource *res, u32 gsi,
 			triggering = trig;
 			polarity = pol;
 		}
-		if (gsi == 5) {
-			polarity = ACPI_ACTIVE_LOW;
-			pr_warning("ACPI: IRQ %d do not override to %s, %s\n", gsi,
-				   t ? "level" : "edge", p ? "low" : "high");
-		}
 	}
 
 	res->flags = acpi_dev_irq_flags(triggering, polarity, shareable);
-- 
2.7.4

