From 141480773be1efc7715b0c519e96e830563fa358 Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Thu, 15 Mar 2018 11:44:44 +0530
Subject: [PATCH 3353/5855] amd-i2c : AMD-MP2-I2C Changed debug print level

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 drivers/i2c/busses/pcie_mp2_amd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/pcie_mp2_amd.c b/drivers/i2c/busses/pcie_mp2_amd.c
index 5df729b..fbfbc94 100755
--- a/drivers/i2c/busses/pcie_mp2_amd.c
+++ b/drivers/i2c/busses/pcie_mp2_amd.c
@@ -79,12 +79,12 @@ int amd_i2c_connect(struct pci_dev *dev,
 	i2c_cmd_base.s.i2c_speed = i2c_connect_config.connection_speed;
 
 	if (i2c_cmd_base.s.bus_id == I2C_BUS_1) {
-		dev_info(ndev_dev(privdata), "%s writing:%x busid:%d reg:%llx\n",
+		dev_dbg(ndev_dev(privdata), "%s writing:%x busid:%d reg:%llx\n",
 				__func__, i2c_cmd_base.ul, i2c_cmd_base.s.bus_id,
 				(u64)privdata->mmio + AMD_C2P_MSG1);
 		writel(i2c_cmd_base.ul, privdata->mmio + AMD_C2P_MSG1);
 	} else if (i2c_cmd_base.s.bus_id == I2C_BUS_0) {
-		dev_info(ndev_dev(privdata), "%s writing:%x at busid:%d reg:%llx\n",
+		dev_dbg(ndev_dev(privdata), "%s writing:%x at busid:%d reg:%llx\n",
 				__func__, i2c_cmd_base.ul, i2c_cmd_base.s.bus_id,
 				(u64)privdata->mmio + AMD_C2P_MSG0);
 		writel(i2c_cmd_base.ul, privdata->mmio + AMD_C2P_MSG0);
-- 
2.7.4

