From 22377109c3033be6c96c18bd6ee2d57cb91ec47b Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Wed, 27 Jan 2016 18:04:20 -0500
Subject: [PATCH 0734/1110] drm/amd/dal: Add DCE10 config option

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Reviewed-by: Jordan Lazare <jordan.lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/Kconfig b/drivers/gpu/drm/amd/dal/Kconfig
index 14df02e..0dc6f86 100644
--- a/drivers/gpu/drm/amd/dal/Kconfig
+++ b/drivers/gpu/drm/amd/dal/Kconfig
@@ -28,6 +28,15 @@ config DRM_AMD_DAL_DCE11_0
 	 CZ family
 	 for display engine
 
+config DRM_AMD_DAL_DCE10_0
+        bool "VI family"
+        depends on DRM_AMD_DAL
+        help
+         Choose this option
+	 if you want to have
+	 VI family for display
+	 engine.
+
 config DEBUG_KERNEL_DAL
         bool "Enable kgdb break in DAL"
         depends on DRM_AMD_DAL
-- 
2.7.4

