From 3eecb0fe3cdece4ea4f3a5dedd523b25af8adf45 Mon Sep 17 00:00:00 2001
From: Arindam Nath <arindam.nath@amd.com>
Date: Mon, 25 Jul 2016 17:56:19 +0530
Subject: [PATCH 1109/1110] drm/amdgpu: fix system randomly reboots after login

Signed-off-by: Arindam Nath <arindam.nath@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/vi.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/vi.c b/drivers/gpu/drm/amd/amdgpu/vi.c
index 7f15c1a..f4027d6 100644
--- a/drivers/gpu/drm/amd/amdgpu/vi.c
+++ b/drivers/gpu/drm/amd/amdgpu/vi.c
@@ -1543,7 +1543,6 @@ static int vi_common_early_init(void *handle)
                                          AMDGPU_PG_SUPPORT_GFX_SMG |
                                          AMDGPU_PG_SUPPORT_GFX_DMG |
                                          AMDGPU_PG_SUPPORT_CP |
-                                         AMDGPU_PG_SUPPORT_RLC_SMU_HS |
                                          AMDGPU_PG_SUPPORT_GFX_PIPELINE;
 
 		adev->external_rev_id = adev->rev_id + 0x1;
-- 
2.7.4

